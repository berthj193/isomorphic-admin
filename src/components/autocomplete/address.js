import React from 'react';
import PropTypes from 'prop-types';
import withOrdersManager from '../apolloProviders/withOrdersManager';
import addressQueries from '../../graphql/addresses';
import Select, { SelectOption } from '../uielements/select';

export class AddressAutocomplete extends React.Component {
  static propTypes = {
    addresses: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      given_name: PropTypes.string,
      family_name: PropTypes.string,
      line_one: PropTypes.string,
      city: PropTypes.string,
    })),
  };

  static defaultProps = {
    addresses: [],
  };

  dataSource = (this.props.addresses || [])
    .map(({ id, given_name, family_name, line_one, city }) =>
      <SelectOption value={id} key={id}>{`${given_name} ${family_name}, ${line_one}, ${city}`}</SelectOption>
    );

  render() {
    return (
      <Select
        showSearch
        {...this.props}
      >
        {this.dataSource}
      </Select>
    );
  }
}

export default withOrdersManager(AddressAutocomplete, [
  {
    query: addressQueries.getAutocompleteList,
  }]);
