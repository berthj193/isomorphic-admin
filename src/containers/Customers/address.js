import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Address extends PureComponent {
  static propTypes = {
    line_one: PropTypes.string,
    line_two: PropTypes.string,
    postcode: PropTypes.string,
  };

  render() {
    const { line_one, line_two, postcode } = this.props;
    return (
      <div className="customer-address">
        <p>{line_one}</p>
        { line_two && <p>{line_two}</p> }
        <p>{postcode}</p>
      </div>
    );
  }
}
