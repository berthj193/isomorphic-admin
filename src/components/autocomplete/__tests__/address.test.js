import React from 'react';
import { AddressAutocomplete } from '../address';
import { shallow } from 'enzyme';

describe('address autocomplete component', () => {
  const getComponent = (props = {}) => shallow(<AddressAutocomplete {...props} />);
  const dataObj = {
    id: '1',
    given_name: 'john',
    family_name: 'doe',
    line_one: 'foo',
    city: 'bar',
  };

  it('datasource should return empty array when there are no elements', () => {
    const instance = getComponent().instance();
    expect(instance.dataSource.length).toBe(0);
  });

  it('datasource should return one element', () => {
    const instance = getComponent({
      addresses: [dataObj],
    }).instance();

    expect(instance.dataSource).toMatchSnapshot();
  });
});
