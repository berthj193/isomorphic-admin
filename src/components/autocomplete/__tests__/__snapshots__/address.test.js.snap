// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`address autocomplete component datasource should return one element 1`] = `
Array [
  <Option
    value="1"
>
    john doe, foo, bar
</Option>,
]
`;
